cscs-checks/prgenv/cuda/cuda_aware_mpi.py

```diff
@@ -0,0 +1,151 @@
+        self.tags={'production', 'external_resources'}
+
+    @ rfm.run_before('compile')
+    def set_compilers(self):
+        if self.current_environ.name == 'PrgEnv-pgi':
+           self.build_system.cflags=['-std=c99', ' -O3']
+
+        self.build_system.options += [
+            'MPICC="%s"' % self.build_system._cc(self.current_environ),
+            'MPILD="%s"' % self.build_system._cxx(self.current_environ)
+        ]
+
+
+class CudaAwareMPIRuns(rfm.RunOnlyRegressionTest):
+    def __init__(self):
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'dom:gpu', 'tiger:gpu',
+                              'arolla:cn', 'tsa:cn',
+                              'ault:amdv100', 'ault:intelv100']
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu']
+        if self.current_system.name in ['arolla', 'tsa', 'daint']:
+            self.valid_prog_environs += ['PrgEnv-pgi']
+        elif self.current_system.name in ['ault']:
+            self.valid_prog_environs = ['PrgEnv-gnu']
+
+        if self.current_system.name in ['arolla', 'tsa','ault']:
+            self.valid_prog_environs = ['PrgEnv-gnu']
+        else:
+            self.modules = ['craype-accel-nvidia60']
+
+        self.executable = '../bin/jacobi_cuda_aware_mpi'
+        self.prerun_cmds = ['export MPICH_RDMA_ENABLED_CUDA=1']
+        self.depends_on('CudaAwareMPICheck')
+        self.sanity_patterns = sn.assert_found(r'Stopped after 1000 iterations'
+                                               r' with residue 0.00024', 
+                                               self.stdout)
+
+ 
+@rfm.simple_test
+class CudaAwareMPIOneNodeCheck(CudaAwareMPIRuns):
+    def __init__(self):
+        super().__init__() 
+        self.partition_num_gpus_per_node = {
+            'daint:gpu':      1,
+            'dom:gpu':        1,
+            'kesh:cn':        2, 
+            'tiger:gpu':      2,
+            'arolla:cn':      2,
+            'tsa:cn':         2,
+            'ault:amdv100':   2, 
+            'ault:intelv100': 4
+        }
+
+        self.prerun_cmds += ['export CRAY_CUDA_MPS=1']
+
+    @rfm.run_before('run')
+    def set_num_gpus_per_node(self):
+        cp = self.current_partition.fullname
+        if cp in self.partition_num_gpus_per_node:
+            self.num_gpus_per_node = self.partition_num_gpus_per_node.get(cp)
+        else:
+            self.num_gpus_per_node = 1
+
+        self.num_tasks = 2 * self.num_gpus_per_node
+        self.num_tasks_per_node = self.num_tasks 
+        self.executable_opts = ['-t %d %d' % (self.num_tasks/2, 2)]
+
+    @rfm.require_deps
+    def set_executable(self, CudaAwareMPICheck):
+        self.executable = os.path.join(
+            CudaAwareMPICheck().stagedir,
+            'posts','cuda-aware-mpi-example',
+            'bin', 'jacobi_cuda_aware_mpi'
+        )
+
+
+@rfm.simple_test
+class CudaAwareMPITwoNodesCheck(CudaAwareMPIRuns):
+    def __init__(self):
+        super().__init__() 
```

Comment:
If I pop this out, the test doesn't run because it won't inherit the `self.valid_systems` and the rest of the settings defined on the parent class `CudaAwareMPIRuns`. This is the same for the `CudaAwareMPIOneNodeCheck`.

Reply:
You are right this is needed :-)